sys/progress_bar/progress_bar.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2019 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_progress_bar
+ * @{
+ *
+ * @file
+ * @brief       Progress bar implementation
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <inttypes.h>
+#include <assert.h>
+
+#include "progress_bar.h"
+
+void progress_bar_update(progress_bar_t *progress_bar)
+{
+    assert(progress_bar->value < 101);
```

Comment:
IMO, chopping at the maximum value would prevent runtime crashes, especially since the assert is compiled out.